### PR TITLE
add service grpc api crate and tests of round-tripping

### PR DESCRIPTION
**Pull request by cbeck88**
add service grpc api crate and tests of round-tripping

(most of the diff here is cargo lock for grpcio addition)

### CODE REVIEW

File: api/proto/grapevine.proto [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+syntax = "proto3";
+
+package grapevine;
+option go_package = "grapevine/api";
+
+import "attest.proto";
+
+service GrapevineAPI {
+    /// This is called to perform IX key exchange with the enclave before calling Query.
```
[the remark this answers is not in the document]
Comment (cbeck88) [reply api/proto/grapevine.proto:11]: Its a mode of noise protocol that we use

File: api/proto/grapevine.proto [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+/// taking into account access patterns and timings of the enclave.
+/// Create is distinguishable from these, when taking into account access patterns and timings.
+message QueryRequest {
+    /// The type of the request. This is one of the integers from RequestType enum.
+    fixed32 request_type = 1;
+    /// A 32 byte ristreto public key corresponding to the identity of the caller.
+    bytes auth_identity = 2;
+    /// A 64 byte ristretto signature, using the private key of the caller, which is
+    /// verified against the auth_identity. This signature is made over 32 random bytes
+    /// drawn from the challenge RNG. (See AuthMessageWithChallengeseed).
```
Comment (eranrund) [inline api/proto/grapevine.proto:108]:
```suggestion
    /// drawn from the challenge RNG. (See AuthMessageWithChallengeSeed).
```

**Review by eranrund — APPROVED** [review]
Nice work documenting the API

File: api/proto/grapevine.proto [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+/// which it uses as a seed to ChaCha20. You must decrypt these bytes after
+/// completing handshake and initializing cipher, and then seed your own ChaCh20.
+/// Whenever you send a query to the grapevine enclave over this channel thereafter,
+/// you must draw 32 bytes from ChaCha20, and sign them with your private key,
+/// to authenticate your identity with the enclave.
+message AuthMessageWithChallengeSeed {
+    /// The auth message normally returned during MobileCoin Auth attestation calls
+    attest.AuthMessage auth_message = 1;
+    /// The .data portion of an EnclaveMessage<ClientSession> from the server to the client.
+    /// The .aad portion is empty, and the session is implied from context.
```
[the remark this answers is not in the document]
Comment (cbeck88) [reply api/proto/grapevine.proto:30]: they actually aren't -- maybe this confusion indicates that i'm doing this in a janky way

the enclave gives an `EnclaveMessage<ClientSession>`, but the `aad` field is always empty, and the client session is inferred from context (we don't need to tell the client their session id, the auth_message contains it), so I just took the `data` field of it, and that's what is returned here by the API. So the way you use it is that you decrypt, and pass `&[]` as the aad.

I could instead make the `encrypted_challenge_seed` be of type `attest::Message` i guess

**Pull request merged**